alicloud/resource_alicloud_slb_ca_certificate.go

```diff
@@ -0,0 +1,133 @@
+package alicloud
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/slb"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudSlbCACertificate() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudSlbCACertificateCreate,
+		Read:   resourceAlicloudSlbCACertificateRead,
+		Update: resourceAlicloudSlbCACertificateUpdate,
+		Delete: resourceAlicloudSlbCACertificateDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: false,
+			},
+			"ca_certificate": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAlicloudSlbCACertificateCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	slbService := SlbService{client}
+
+	request := slb.CreateUploadCACertificateRequest()
+
+	if val, ok := d.GetOk("name"); ok && val.(string) != "" {
+		request.CACertificateName = val.(string)
+	}
+
+	if val, ok := d.GetOk("ca_certificate"); ok && val.(string) != "" {
+		request.CACertificate = val.(string)
+	} else {
+		return fmt.Errorf("UploadCACertificate got an error, ca_certificate should be not null")
+	}
+
+	raw, err := slbService.client.WithSlbClient(func(slbClient *slb.Client) (interface{}, error) {
+		return slbClient.UploadCACertificate(request)
+	})
+	if err != nil {
+		return fmt.Errorf("UploadCACertificate got an error: %#v", err)
+	}
+	response := raw.(*slb.UploadCACertificateResponse)
+
+	d.SetId(response.CACertificateId)
+
+	return resourceAlicloudSlbCACertificateUpdate(d, meta)
+}
+
+func resourceAlicloudSlbCACertificateRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	slbService := SlbService{client}
+
+	caCertificate, err := slbService.describeSlbCACertificate(d.Id())
+	if err != nil {
+		if IsExceptedError(err, SlbCACertificateIdNotFound) || NotFoundError(err) {
+			d.SetId("")
```

Comment:
I think `IsExceptedError(err, SlbCACertificateIdNotFound)` is redundant and please remove it.

Reply:
yes, it's redundant.